tests/cpp/multidevice_transformer.cpp

```diff
@@ -0,0 +1,660 @@
+// clang-format off
```

Comment:
This file is almost directly copy paste from `test_multidevice_transformer.cpp`
There was some slight clean up in merging Forward and Forward_SP test into one parameterized function to reduce (some) duplication.